ndcube/utils/wcs.py

```diff
@@ -197,18 +197,30 @@ def _wcs_slicer(wcs, missing_axis, item):
                 item_checked.append(slice(0, 1))
         # if all are slice in the item tuple
         if _all_slice(item_checked):
-            new_wcs = wcs.slice((item_checked))
+            item_ = (item_checked)
         # if all are not slices some of them are int then
         else:
             # this will make all the item in item_checked as slice.
             item_ = _slice_list(item_checked)
-            new_wcs = wcs.slice(item_)
             for i, it in enumerate(item_checked):
                 if isinstance(it, int):
                     missing_axis[i] = True
     # returning the reverse list of missing axis as in the item here was reverse of
     # what was inputed so we had a reverse missing_axis.
-    return new_wcs, missing_axis[::-1]
+    extra_coords_list = [] # initiating new list
+    for slice_element in item_: # check item_ slices for dropped axes if any
+        for _bool in missing_axis:
+            if not _bool:
+                if index is not 1: 
+                    if (slice_element.stop - slice_element.start) == 1:
+                        px, py = slice_element.all_pix2world(slice_element.wcs.crpix[0], slice_element.wcs.crpix[1], 1)
+                            if px == None:
```

Comment:
Sorry, this bit was really weird-looking now that I realized what I did wrong... Have removed it since I added in the suggested `pix_coords`, as in the following which I quote:
```
pix_coords = [np.array([0])] * len(item_) 
pix_coords[i][0] = slice_element.start 
real_world_coords = wcs.all_pix2world(*pix_coords, 0)
```

Reply:
Hi @DanRyanIrish, I would like to ask if using the desired `all_pix2world` transformation on a `wcs` object to obtain the coordinate values necessarily requires that my input `pix_coords` as defined be [X, Y] pairs?... Just wanted to make sure.

Reply:
Here is the documentation describing the inputs for ```all_pix2world```.  http://docs.astropy.org/en/stable/api/astropy.wcs.WCS.html#astropy.wcs.WCS.all_pix2world

I find the "more than two arguments" format more intuitive.  In ```*pix_coords``` the ```*``` unpacks the values in ```pix_coords``` as though you entered them as separate arguments.  The input arrays within ```pix_coords``` actually describe the [x_values, y_values, z_values, etc.] depending on the number of dimensions.  In this way you can convert multiple pixels to real world coordinates at once.  However, in this case we are only looking for a single pixel's value so the format becomes equivalent to [X, Y, Z, etc.] "pairs".

Reply:
Thanks, @DanRyanIrish, now I see. It is indeed more intuitive to think of `pix_coords` as multi-dimensional now that I have time to let the idea sink in.